### PR TITLE
Added workaround for ios safari iframe issue

### PR DESCRIPTION
**Pull request by nbayati**
Created a workaround for ios Safari by adding a dummy event listener.

Fixes: https://github.com/flutter/flutter/issues/70858
Fixes: https://github.com/flutter/flutter/issues/101251

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/web_ui/lib/src/engine/pointer_binding.dart [hunk 1]
```diff
@@ -62,6 +62,14 @@ int convertButtonToButtons(int button) {
   }
 }
 
+class SafariPointerEventWorkaround {
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/pointer_binding.dart:65]: Consider documenting public classes and methods.

File: lib/web_ui/lib/src/engine/browser_detection.dart [hunk 2]
```diff
@@ -236,6 +236,11 @@ bool get isIOS15 {
       html.window.navigator.userAgent.contains('OS 15_');
 }
 
+/// A flag to check if the browser is iOS Safari.
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/browser_detection.dart:239]: I know in other places we referred to getters like this as "flag" but typically a "flag" refers to a boolean environment variable or a compiler option. I'd just start with "Returns true if ..."

File: lib/web_ui/test/engine/pointer_binding_test.dart [hunk 3]
```diff
@@ -2241,6 +2241,15 @@ void testMain() {
   );
 }
 
+class MockSafariPointerEventWorkaround implements SafariPointerEventWorkaround {
```
Comment (yjbanov) [inline lib/web_ui/test/engine/pointer_binding_test.dart:2244]: I don't see this class being used.

**Review by mdebbar — COMMENTED** [review]
There seem to be some build errors:
```
error - test/canvaskit/semantics_test.dart:28:12 - Undefined name 'isIosSafari'. Try correcting the name to one that is defined, or defining the name. - undefined_identifier
error - test/engine/semantics/semantics_test.dart:18:8 - Target of URI doesn't exist: '../../common.dart'. Try creating the file referenced by the URI, or Try using a URI for a file that does exist. - uri_does_not_exist
```

File: lib/web_ui/test/engine/pointer_binding_test.dart [hunk 4]
```diff
@@ -38,7 +38,8 @@ void _testEach<T extends _BasicEventContext>(
 
 /// Some methods in this class are skipped for iOS-Safari.
 // TODO(mdebbar): https://github.com/flutter/flutter/issues/60033
-bool get isIosSafari => browserEngine == BrowserEngine.webkit &&
+bool get isIosSafari =>
+    browserEngine == BrowserEngine.webkit &&
     operatingSystem == OperatingSystem.iOs;
```
[the remark this answers is not in the document]
Comment (nbayati) [reply lib/web_ui/test/engine/pointer_binding_test.dart:43]: Yeah I think you're right.

**Comment by skia-gold** [comment]
Gold has detected about 8 new digest(s) on patchset 15.
View them at https://flutter-engine-gold.skia.org/cl/github/32415

**Comment by mdebbar** [comment]
It looks like the cleanup didn't fix the failure. I spent some time this morning debugging the test failure and found the problem. The cleanup that you did is necessary, but it's not working because the listeners are stored as static members:
https://github.com/flutter/engine/blob/1d0c2ae6fd0dabc43a2dc41b4053136161097374/lib/web_ui/lib/src/engine/pointer_binding.dart#L161-L165

That means all PointerBinding/Adapter instances are sharing the same map of listeners.

If you remove `static` from those two members, things will start working as expected.

**Review by mdebbar — APPROVED** [review]
LGTM!

**Pull request merged**